test/functional/runner/index.js

```diff
@@ -428,6 +410,50 @@ function normalizeReadPreference(mode) {
   return mode.charAt(0).toLowerCase() + mode.substr(1);
 }
 
+function resolveOperationArgs(operationName, operationArgs, context) {
+  const result = [];
+  function pluck(fromObject, toArray, fields) {
+    for (const field of fields) {
+      if (fromObject[field]) toArray.push(fromObject[field]);
+    }
+  }
+
+  // TODO: migrate all operations here
+  if (operationName !== 'distinct') {
```

Comment:
to make this easier to work, perhaps we should switch this to something like:

```js
if (opertionName === 'distinct') {
  ...
} else {
  return;
}
```

or:

```js
switch(operationName) {
  case 'distinct':
    ...
    break;
  default:
    return;
}
```

Reply:
done. Again, I'm not sure this is the best approach, we might be better off creating "models" for each operation in the future - this just works today.